### PR TITLE
Adding tests for YIELD, SKIP, and LIMIT

### PR DESCRIPTION
**Pull request by WWW0030**
# Description

Added edge case testing for newly implemented features YIELD SKIP and LIMIT

# Contributor agreement

- [ ] I have read and agree to the [Contributor Agreement](https://github.com/kuzudb/kuzu/blob/master/CLA.md).

### CODE REVIEW

File: test/test_files/call/call.test [hunk 1]
```diff
@@ -292,6 +292,41 @@ JSON|Adds support for JSON operations
...
+NULL|False
+
+-STATEMENT call table_info('Person') YIELD `property id`, `name`, `type`, `default expression` as `de`, `primary key` as pk RETURN `pk`, `name`;
+---- 3
+True|id
+False|name
+False|age
+
+-STATEMENT call table_info('Person') YIELD `property id`, `name`, `type` RETURN `name`;
+Binder exception: Output variables must all appear in the yield clause.
```
Comment (acquamarin) [inline test/test_files/call/call.test:328]: We already have such test cases, why add them again?

File: test/test_files/copy/copy_with_skip_lines.test [hunk 2]
```diff
@@ -2,6 +2,55 @@
 
 --
 
+-CASE LoadWithDoubleSkip
+-STATEMENT LOAD FROM '${KUZU_ROOT_DIRECTORY}/dataset/tinysnb/vPerson.csv' (skip=3) RETURN * ORDER BY age SKIP 1;
```
Comment (acquamarin) [inline test/test_files/copy/copy_with_skip_lines.test:6]: Can you only add the test cases which we don't have?

**Review by acquamarin — CHANGES_REQUESTED** [review]

File: test/test_files/call/call.test [hunk 3]
```diff
@@ -292,6 +292,38 @@ JSON|Adds support for JSON operations
...
+1|name|STRING|NULL|False
+2|age|INT64|NULL|False
+
+-STATEMENT call table_info('PersonTableInfo') YIELD `property id`, `name`, `type`, `default expression`, `primary key` RETURN `default expression`, `primary key`
+---- 3
+NULL|True
+NULL|False
+NULL|False
+
+-STATEMENT call table_info('PersonTableInfo') YIELD `property id`, `name`, `type`, `default expression` as `de`, `primary key` as pk RETURN `default expression`, `primary key`
```
Comment (acquamarin) [inline test/test_files/call/call.test:311]: Ditto, why add those tests?

File: test/test_files/call/call.test [hunk 4]
```diff
@@ -292,6 +292,38 @@ JSON|Adds support for JSON operations
 POSTGRES|Adds support for reading from POSTGRES tables
 SQLITE|Adds support for reading from SQLITE tables
 
+-CASE CallTableInfoWithYield
+-STATEMENT CREATE NODE TABLE PersonTableInfo(id INT64 PRIMARY KEY, name STRING, age INT64)
+---- ok
+
+-STATEMENT CALL table_info('PersonTableInfo') YIELD `property id`, `name`, `type`, `default expression`, `primary key` RETURN *
```
Comment (acquamarin) [inline test/test_files/call/call.test:299]: We already had a test that calls table_info on a table and yield every field.
Why do you add them again?

File: test/test_files/copy/copy_with_skip_lines.test [hunk 5]
```diff
@@ -2,6 +2,46 @@
...
+-CASE LoadWithExpressionSkip
+-STATEMENT LOAD FROM '${KUZU_ROOT_DIRECTORY}/dataset/tinysnb/vPerson.csv' RETURN * SKIP 4.1;
+---- error
+Runtime exception: The number of rows to skip/limit must be a non-negative integer.
+
+-STATEMENT LOAD FROM '${KUZU_ROOT_DIRECTORY}/dataset/tinysnb/vPerson.csv' RETURN * SKIP -4;
+---- error
+Runtime exception: The number of rows to skip/limit must be a non-negative integer.
+
+-STATEMENT LOAD FROM '${KUZU_ROOT_DIRECTORY}/dataset/tinysnb/vPerson.csv' RETURN * SKIP 'abc';
```
Comment (acquamarin) [inline test/test_files/copy/copy_with_skip_lines.test:14]: We also have such case under skip/limit.test
Can you double check?

File: test/test_files/call/call.test (hunk 4 above)
Comment (WWW0030) [reply test/test_files/call/call.test:299]: I don't think we have a test that calls table_info and yield every field, we have one for show_connection() though, so I think we should keep it in

File: test/test_files/copy/copy_with_skip_lines.test (hunk 5 above)
Comment (WWW0030) [reply test/test_files/copy/copy_with_skip_lines.test:14]: Yes, but the skip tests I added are not tested in the skip_limit.test file. I will move it to that file instead 👍

File: test/test_files/projection/skip_limit.test [hunk 6]
```diff
@@ -2,6 +2,46 @@
 
 --
 
+-CASE LoadWithExpressionSkip
+-STATEMENT LOAD FROM '${KUZU_ROOT_DIRECTORY}/dataset/tinysnb/vPerson.csv' RETURN * SKIP 4.1;
+---- error
+Runtime exception: The number of rows to skip/limit must be a non-negative integer.
```
Comment (acquamarin) [inline test/test_files/projection/skip_limit.test:8]: These tests already exist in binder_error.test:
```
-LOG InvalidLimitNumberType
-STATEMENT MATCH (a:person) RETURN a.age LIMIT "abc"
---- error
Runtime exception: The number of rows to skip/limit must be a non-negative integer.

-LOG NegativeLimitNumber
-STATEMENT MATCH (a:person) RETURN a.age LIMIT -1
---- error
Runtime exception: The number of rows to skip/limit must be a non-negative integer.

```

File: test/test_files/projection/skip_limit.test [hunk 7]
```diff
@@ -2,6 +2,46 @@
...
+
+-STATEMENT LOAD FROM '${KUZU_ROOT_DIRECTORY}/dataset/tinysnb/vPerson.csv' RETURN * SKIP -4;
+---- error
+Runtime exception: The number of rows to skip/limit must be a non-negative integer.
+
+-STATEMENT LOAD FROM '${KUZU_ROOT_DIRECTORY}/dataset/tinysnb/vPerson.csv' RETURN * SKIP 'abc';
+---- error
+Runtime exception: The number of rows to skip/limit must be a non-negative integer.
+
+-STATEMENT LOAD FROM '${KUZU_ROOT_DIRECTORY}/dataset/tinysnb/vPerson.csv' RETURN * SKIP 2*2;
```
Comment (acquamarin) [inline test/test_files/projection/skip_limit.test:18]: What is the difference between your test and the one in this file?
```
-LOG SkipWithConstantExpr
-STATEMENT MATCH (a:person) RETURN a.ID SKIP 1 + 2
---- 5
5
7
8
9
10
```
Both of them are testing constant expression as skip number

File: test/test_files/projection/skip_limit.test [hunk 8]
```diff
@@ -2,6 +2,46 @@
...
+---- 1
+7|Elizabeth|1|False|True|20|4.7|1980-10-26|1976-12-23 11:21:42|48:24:11|[2]|[Ein]|[[6],[7],[8]]|[96,59,65,88]|1.463|{a0eebc99-9c0b4ef8-bb6d6bb9-bd380a15}
+
+-STATEMENT CREATE NODE TABLE PersonLoad(id INT64 PRIMARY KEY);
+---- ok
+
+-STATEMENT CREATE (n:PersonLoad {id:1});
+---- ok
+
+-STATEMENT MATCH(p:PersonLoad) LOAD FROM '${KUZU_ROOT_DIRECTORY}/dataset/tinysnb/vPerson.csv' RETURN * SKIP p.id;
```
Comment (acquamarin) [inline test/test_files/projection/skip_limit.test:28]: This is also covered, do search the tests before adding a new one

File: test/test_files/projection/skip_limit.test [hunk 9]
```diff
@@ -2,6 +2,46 @@
...
+---- error
+Binder exception: The number of rows to skip/limit must be a parameter/literal expression.
+
+-STATEMENT LOAD FROM '${KUZU_ROOT_DIRECTORY}/dataset/tinysnb/vPerson.csv' RETURN * SKIP CAST(4.6, "INT8");
```
Comment (acquamarin) [inline test/test_files/projection/skip_limit.test:32]: These casting tests are fine, you can keep them

File: test/test_files/call/call.test (hunk 4 above)
Comment (acquamarin) [reply test/test_files/call/call.test:299]: The test has no difference with  calling `show_connection()` and yield every field.
We don't have to replicate those tests

File: test/test_files/projection/skip_limit.test (hunk 8 above)
Comment (WWW0030) [reply test/test_files/projection/skip_limit.test:28]: ok, will do in the future. 👍

**Review by acquamarin — APPROVED** [review]
